app-modules/interaction/tests/Models/Scopes/InteractionConfidentialScopeTest.php

```diff
@@ -35,55 +35,92 @@
 */
 
 use AdvisingApp\Authorization\Enums\LicenseType;
-use AdvisingApp\Interaction\Filament\Resources\InteractionResource\Pages\ListInteractions;
 use AdvisingApp\Interaction\Models\Interaction;
+use AdvisingApp\Interaction\Models\Scopes\InteractionConfidentialScope;
 use AdvisingApp\Team\Models\Team;
 use App\Models\User;
 
 use function Pest\Laravel\actingAs;
-use function Pest\Livewire\livewire;
 use function Tests\asSuperAdmin;
 
-test('ListInteration with display data using the is_confidential field', function () {
+test('Interaction model has applied global scope', function () {
+    Interaction::factory()->create();
+
+    expect(Interaction::hasGlobalScope(InteractionConfidentialScope::class))->toBeTrue();
+});
+
+test('Interactions model with fetch data count for created user', function () {
     $user = User::factory()->licensed(LicenseType::cases())->create();
     $user->givePermissionTo('interaction.view-any');
 
-    $userWithoutAttach = User::factory()->licensed(LicenseType::cases())->create();
-    $userWithoutAttach->givePermissionTo('interaction.view-any');
+    actingAs($user);
+    Interaction::factory()->count(10)->create([
+        'is_confidential' => true,
+        'user_id' => $user,
+    ]);
+
+    Interaction::factory()->count(10)->create([
+        'is_confidential' => true,
+    ]);
 
+    Interaction::factory()->count(10)->create([
+        'is_confidential' => false,
+    ]);
+
+    expect(Interaction::query()->count())->toBe(20);
+});
+
+test('Interactions model with fetch data count for team user', function () {
     $teamUser = User::factory()->licensed(LicenseType::cases())->create();
     $teamUser->givePermissionTo('interaction.view-any');
```

Comment:
I don't thin you need to give this permission is this test or any of the others, right? The global scope doesn't check permission. Just if super admin.